### PR TITLE
upgrading dockest to 2.1.0

### PR DESCRIPTION
**Pull request by dskatz22**
In this PR
- Upgrading dockest to 2.1.0 to support a more flexible integration testing suite wrt configuring the generated docker-compose file. Version 2.1.0 uses the docker-compose file as a base for the integration test suite, instead of having pre-configured runners. 

Additional Notes
One use case that came to mind as I was testing [PR](https://github.com/kafkajs/confluent-schema-registry/pull/93) that it might be nice to add other confluent producers to the integration testing suite to test that the deserializers work with other components.

### CODE REVIEW

**Comment by dskatz22** [comment]
@Nevon would you happen to have any bandwidth to take a look at this small-ish PR?

**Comment by Nevon** [comment]
This is really an @erikengervall thing, as he's the author of Dockest, but I'll try to make time for this once I finish up #93.

**Comment by Nevon** [comment]
Got some conflicts to resolve, but otherwise it seems fine. As long as the tests are passing, it should be good to go.

File: docker-compose.yml [hunk 1]
```diff
@@ -32,3 +32,5 @@ services:
 
   avro_tools: # https://hub.docker.com/r/coderfi/avro-tools
     image: coderfi/avro-tools:1.7.7
+    ports:
+     - '9999:9999'
```
Comment (Nevon) [inline docker-compose.yml:36]: Why do we need to expose this port for avro-tools?

Comment (dskatz22) [reply docker-compose.yml:36]: There is a bug in dockest I believe where the docker-compose must define a port.  I'll look into the root cause and see if there is a workaround that doesn't require this hack.

Comment (dskatz22) [reply docker-compose.yml:36]: @erikengervall is this something I should address in dockest, or is there a known workaround?

Comment (erikengervall) [reply docker-compose.yml:36]: > @erikengervall is this something I should address in dockest, or is there a known workaround?

Hey! There's even a version 3.0.0 of Dockest in the making which will include a number of improvements. Can't tell exactly what the ETA is, but there's a few PRs queued to be merged and some beta testing required before we can release.

> I'll look into the root cause and see if there is a workaround that doesn't require this hack.

I'd say this port-hack is okay for now, I could circle back to this piece of code once Dockest v3 is out and this has been addressed 🙂

**Review by erikengervall — APPROVED** [review]

**Pull request merged**